File: kotlin/src/test/kotlin/Number of 1 Bits Test.kt
import o.mg.hammingWeight
import kotlin.test.Test
import kotlin.test.assertEquals

class `Number of 1 Bits Test` {
    @Test
    fun case1() {
        val result = hammingWeight(11)
        assertEquals(3, result)
    }

    @Test
    fun case2() {
        val result = hammingWeight(128)
        assertEquals(1, result)
    }

    @Test
    fun case3() {
        val result = hammingWeight(2147483645)
        assertEquals(30, result)
    }
}

File: go/solutions/test/number_of_1_bits_test.go
package solutions_test

import (
	"reflect"
	"solutions"
	"testing"
)

func TestHammingWeight(t *testing.T) {
	type args struct {
		n int
	}
	tests := []struct {
		name     string
		args     args
		expected int
	}{
		{
			"case1",
			args{11},
			3,
		},
		{
			"case2",
			args{128},
			1,
		},
		{
			"case3",
			args{2147483645},
			30,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := solutions.HammingWeight(tt.args.n); !reflect.DeepEqual(got, tt.expected) {
				t.Errorf("HammingWeight(%v) = %v, want %v", tt.args.n, got, tt.expected)
			}
		})
	}
}
